kratos/containers/array_1d.h

```diff
@@ -761,4 +768,34 @@ class	array_1d	: public boost::numeric::ublas::vector_expression< array_1d<T, N>
 
 }  // namespace	Kratos.
 
-#endif // KRATOS_ARRAY_1D_H_INCLUDED  defined
+namespace std
+{
+namespace 
+{
+    /**
+     * @brief This method creates an "unique" hash for the input value
+     * @tparam TClassType The type of class to be hashed
+     * @param rSeed This is the seed used to create the hash
+     * @param rValue This is the value to be hashed
+     * @todo Once the hashers and comparors are moved from key_hash.h, include key_hash and remove this. Right now there is a cross inclussion
+     */
+    template <class TClassType>
+    inline void HashCombine(
+        std::size_t& rSeed,
+        const TClassType& rValue
+        )
+    {
+        std::hash<TClassType> hasher;
+        rSeed ^= hasher(rValue) + 0x9e3779b9 + (rSeed<<6) + (rSeed>>2);
```

Comment:
this smells like a copy-pasta from stackoverflow. Can you please elaborate on what this does and why these magic numbers are chosen?

Reply:
It is a copy paste from my own code, which probably I took from stackoverflow

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/1f89ad0f5ab9b554f80013bd0bd39077cd083a6e/kratos/includes/key_hash.h#L57

Reply:
It is boost has_combine: https://stackoverflow.com/questions/35985960/c-why-is-boosthash-combine-the-best-way-to-combine-hash-values

Reply:
(used in Kratos since several years ago)

Reply:
Oh and please leave a comment that this impl comes from boost.

Reply:
Okay (this means more compilation time...)

Reply:
> Oh and please leave a comment that this impl comes from boost.

Yes, as I said I just now rediscover, at that time I just copied from stackoverflow

![image](https://github.com/KratosMultiphysics/Kratos/assets/19991680/62e0b2a9-3d4f-4c1f-958b-9535751f472e)

Reply:
I copied from here (Firefox still remembers): https://stackoverflow.com/questions/7222143/unordered-map-hash-function-c

Reply:
I don't have a problem with copying a snippet, but those magic numbers are a bit unnerving without proper explanation/sources.

Reply:
> I copied from here (Firefox still remembers): https://stackoverflow.com/questions/7222143/unordered-map-hash-function-c

The [primary source](https://www.boost.org/doc/libs/1_55_0/doc/html/hash/reference.html#boost.hash_combine) would be better.

Reply:
If you take into account that I introduced in Kratos 6 years ago when Kratos was still moving from subversion and everything was super chaotic and documentation was close to 0 (closer than now), we are lucky that I can trackback my own web history and see what I did instead of being unknown legacy code

Reply:
Done @matekelemen